Use vector drawables instead of PNGs for material icons

Stypox opened the pull request:
### What is it?
- [ ] Bug fix
- [ ] Feature
- [x] Enhancement under the hood

### Long description of the changes in your PR
For all manually-created images PNG have been kept. Debug apk size decreased by 232,030 bytes.
- Rename all icon attrs to have a **`ic_` prefix**
- Always use **`_24dp` icons**, because there is no real difference, since they are vector drawables
- Always use the **original name found on material.io for icon** drawables, as to not create confusion and possibly duplicates. Icon names can still be different from real drawable names, though I have made some of them compliant to this or maybe more meaningul.
- Use standard icons for **`expand_more` and `expand_less`** instead of triangles
- Use `play_button_outline` instead of custom PNG as **play button in VideoDetailFragment** (questionable, as there is no shadow anymore)
- **Replace the search-insert-icon** (on the button that is right to every search suggestion, and when pressed replaces what is in the search field with the value of that suggestion), that was an arrow pointing up-left and didn't have a meaning, with a custom hand-made icon (`ic_search_add`)
- Keep `ic_close` and `ic_replay` as PNGs: needed for notifications in RemoteViews

Other changes:
- Remove duplicate `getIconByAttr()` in ThemeHelper (use `resolveResourceIdFromAttr()` instead)
- Improve the **VideoDetailFragment layout** by a little by forcing the "expand description" icon to always stay in the same place even after expanding. I tested the new layout on various screens and it always looked ok.
- **Update AndroidX** to `1.1.0` to fix icon crashes in preferences on API 19: use `androidx.preference:preference` instead of `androidx.legacy:legacy-preference-v14`
- **Remove legacy libraries** (`androidx.legacy:legacy-support-v4`): this made no effect, and allowed me not to use `androidx.appcompat.widget.*` everywhere, tested on various Android versions
- Fix `SelectChannelFragment` and `SelectKioskFragment` (those used to set main screen tabs): in API 19 they had a **duplicate title** always empty.
- Fix pause icon used instead of play when seeking in a paused popup player

### Fixes the following issue(s)
- Fixes #1199

### Testing apk
Tested thoroughly on API 19 (Galaxy Nexus (small screen) and Pixel 3 XL (big screen) emulators on Android 4.4), on API 24 (native on Huawei P9 Lite with Android 7.0), on API 29 (Pixel 3 XL emulator on Android 10). Though I'm not entirely sure I have explored every single piece of the app, so I'd like @Domiuns and @AnotherLife to test it, too, on Android 4.4 ;-)
Debug apk: [app-debug_2.zip](https://github.com/TeamNewPipe/NewPipe/files/4419662/app-debug_2.zip)


### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

B0pol commented on app/src/main/java/org/schabi/newpipe/player/ServicePlayerActivity.java:
```diff
@@ -700,11 +699,9 @@ private void onMaybeMuteChanged() {
-            Drawable drawableUnmuted = getResources().getDrawable(R.drawable.ic_volume_off_gray_24dp);
-            item.setIcon(player.isMuted() ? drawableMuted : drawableUnmuted);
+            item.setIcon(player.isMuted()
+                    ? getResources().getDrawable(ThemeHelper.resolveResourceIdFromAttr(getApplicationContext(), R.attr.ic_volume_off))
```
I wasn't aware of ThemeHelper.resolveResourceIdFromAttr (and it seems like the one who first put the button, and the reviewers too),
you should rebase, because it [was changed yesterday](https://github.com/TeamNewPipe/NewPipe/pull/3252/files#), now both muted and unmuted use theme.

Stypox replied:
Done, is it ok now?

B0pol replied:
I haven't tested yet but the code seems ok

Stypox replied:
I tested and it worked as expected. Though for now do not test this PR as it is not ready ;-)

AnotherLife commented:
Oh, no, it crashed when I tried to fast forward with double-tapping. While in fullscreen.

https://pastebin.com/QmfEPdsf

Stypox commented:
@AnotherLife fixed with the latest commit, could you test this new apk? [app-debug_1.zip](https://github.com/TeamNewPipe/NewPipe/files/4419222/app-debug_1.zip)

AnotherLife commented:
Seems good now :D

AnotherLife commented:
Wait, there is no icon to maximise a popup window! The control is there and it works but there is no icon visible.

Stypox commented:
I fixed it in the last commit, but didn't upload a new apk due to connection problems 😅

Stypox commented:
Here it is: [app-debug_2.zip](https://github.com/TeamNewPipe/NewPipe/files/4419662/app-debug_2.zip)

Stypox commented:
@TobiGr @wb9688 can I add this to the 0.19.3 milestone? It should be reviewed&merged soon imo, because there are many changed files causing conflicts and at the end of the day it is not really a breaking change.

wb9688 commented:
@Stypox: We don't want to add anything new to the v0.19.3 milestone. I think you should add it to the v0.20.0 milestone.

Stypox commented:
I just rebased

Stypox commented:
Rebased again, this time it was a mess because of the whitespace changes @wb9688 made. Can this be considered soon?

wb9688 commented:
@Stypox: Both `fragment_video_detail.xml`s aren't valid now…

We should test this as soon as possible and merge it.

Stypox commented:
Fixed the rebasing, squashed some commits and checked that now it builds.

wb9688 commented:
I think you could use a [`MaterialShapeDrawable`](https://developer.android.com/reference/com/google/android/material/shape/MaterialShapeDrawable) to add a shadow to the play button.

Stypox commented:
> I think you could use a MaterialShapeDrawable to add a shadow to the play button.

Ok, I will try experimenting with that

> Also, the icon after search suggestions is weird.

In my opinion it is more meaningful than the one used before (it did not even catch my attention once, before doing this pr, because an arrow pointing top-left meant nothing). I can revert this if needed, but I prefer the new icon, even though I agree it is a little bit weird

wb9688 commented:
@Stypox: Well, basically every app with search has the arrow pointing top-left, so…

AnotherLife commented:
Top left arrow means putting the suggestion text into the text box, everyone understands that, for example ... Google uses that!

TobiGr commented:
yes, we should keep the old icon

Stypox commented:
I reverted the change of the search_add drawable to an arrow pointing top-left (i.e. the [ic_call_made](https://material.io/icons/#ic_call_made) icon but mirrored, see my answer [here](https://ux.stackexchange.com/questions/112605)).
Unfortunately `MaterialShapeDrawable` can not be used with arbitrary shapes but only with rectangles, so this way wasn't viable. So I added a circular gradient as background to the ic_play_button_outline drawable (the new file is ic_play_button_shadow_white_24dp.xml). I did the same with the headset icon for audio streams (where the play arrow is replaced by a headset).
I tested again on Android 4.4 and everything works.

wb9688 commented:
@Stypox: That new shadow you added doesn't look amazing, but it's better than no shadow.

TobiGr commented on app/src/main/res/drawable/ic_headset_shadow_white_24dp.xml:
```diff
@@ -0,0 +1,28 @@
+                    android:gradientRadius="12"
+                    android:type="radial">
+                    <item
+                        android:color="#4C000000"
```
```suggestion
                        android:color="#90000000"
```

TobiGr commented on app/src/main/res/drawable/ic_play_arrow_shadow_white_24dp.xml:
```diff
@@ -0,0 +1,28 @@
+                    android:gradientRadius="11"
+                    android:type="radial">
+                    <item
+                        android:color="#4C000000"
```
```suggestion
                        android:color="#90000000"
```

TobiGr commented:
To be honest, I'd like to keep the old icon. Unfortunately, this is only partially possible, because the filters used to create the shadows are not supported by Android vector drawables. To keep the contrast, I added a thin stroke to the icon.
[ic_new_play_arrow_thick.xml.txt](https://github.com/TeamNewPipe/NewPipe/files/4638633/ic_new_play_arrow_thick.xml.txt)

![image](https://user-images.githubusercontent.com/17365767/82123770-6812a980-979b-11ea-8a2d-960e1d35c86c.png)

TobiGr commented on app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java:
```diff
@@ -399,8 +399,8 @@ private void initToolbar(final Toolbar toolbar) {
         boolean isLight = ThemeHelper.isLightThemeSelected(getActivity());
```
that var is unused now.
```suggestion

```

TobiGr commented on app/src/main/res/layout-large-land/fragment_video_detail.xml:
```diff
@@ -485,7 +486,7 @@
                             android:clickable="true"
                             android:contentDescription="@string/controls_download_desc"
-                            android:drawableTop="?attr/download"
+							app:drawableTopCompat="?attr/ic_file_download"
```
You've used tabs in some files

Stypox commented:
These currently are the files with tabs. I'm going to replace everything with spaces:
```
$ grep -rnwl --include=\*.{java,xml} --exclude={\strings.xml,\values-*.xml,\merger.xml} './' -P -e '\t'
./app/src/main/res/drawable/progress_youtube_horizontal_dark.xml
./app/src/main/res/drawable/progress_soundcloud_horizontal_dark.xml
./app/src/main/res/drawable/progress_youtube_horizontal_light.xml
./app/src/main/res/drawable/progress_soundcloud_horizontal_light.xml
./app/src/main/res/layout/list_stream_grid_item.xml
./app/src/main/res/layout/missions.xml
./app/src/main/res/layout/mission_item_linear.xml
./app/src/main/res/layout/list_stream_playlist_grid_item.xml
./app/src/main/res/layout/mission_item.xml
./app/src/main/res/layout/list_playlist_grid_item.xml
./app/src/main/res/menu/dialog_url.xml
```

wb9688 commented:
@Stypox: Android Studio has a button to do that

Stypox commented:
Done, now tabs should have been replaced by spaces in every file (the only ones I excluded are `.svg` assets, but those are files generated by a program). I replaced the play-button with @TobiGr's one and also did the same to the background-play-button (i.e. ic_headset_shadow), so that SoundCloud videos don't look strange.

<img width="300px" src="https://user-images.githubusercontent.com/36421898/82148193-19c0e180-9853-11ea-9d7e-78be0b6e4af3.png"/>

AnotherLife commented:
I have a Kitkat tablet, is there an apk I can test?

Stypox commented:
@AnotherLife @Domiuns @mqus @jojoxl01 @ask6155 @skil3z could you test this apk on your Android 4.4 devices? In particular you sould verify if all icons are visible. Thank you in advance :-)
[app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/4658732/app-debug.zip)

AnotherLife commented:
Looks good to me, I didn't see any icons missing. Unless I didn't see them because they were missing ... *badum tisshhh*

TobiGr commented:
Ah. just saw one thing: Please undo the icon change in the navigation drawer. According to [material.io](https://material.io/components/navigation-drawer) a:small_red_triangle_down: should be used and not a  `v`.

Stypox commented:
Oh sorry, didn't notice that. I replaced the expand icon with a drop one in the latest commit:
[app-debug_1.zip](https://github.com/TeamNewPipe/NewPipe/files/4660987/app-debug_1.zip)

jojoxl01 commented:
At present, no problems have been found and everything is normal. Thank you!

------------------------------

在 2020年5月21日星期四，Stypox <notifications@github.com> 写道：

> Oh sorry, didn't notice that. I replaced the expand icon with a drop one
> in the latest commit:
> app-debug_1.zip
> <https://github.com/TeamNewPipe/NewPipe/files/4660987/app-debug_1.zip>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/TeamNewPipe/NewPipe/pull/3271#issuecomment-631924901>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHZ5MWIQS5IIC3ORM3OFBMLRSTHZHANCNFSM4LTRDG3Q>
> .
>
